src/pages/Graph/SummaryPanelGraph.tsx

```diff
@@ -95,31 +105,101 @@ export default class SummaryPanelGraph extends React.Component<SummaryPanelPropT
-            {this.shouldShowRPSChart() && (
-              <div>
-                <hr />
-                {this.renderRpsChart()}
-              </div>
-            )}
-          </div>
+        <div
+          className="panel-body"
+          style={{ padding: '0px', paddingLeft: '15px', paddingRight: '15px', paddingBottom: '15px' }}
+        >
+          <TabContainer id="basic-tabs" defaultActiveKey="incoming">
+            <div>
+              <Nav className={`nav nav-tabs nav-tabs-pf ${panelNavTabs}`}>
+                <NavItem eventKey="incoming">
+                  <div>Incoming </div>
+                </NavItem>
+                <NavItem eventKey="outgoing">
+                  <div>Outgoing </div>
+                </NavItem>
+                <NavItem eventKey="total">
+                  <div>Total</div>
+                </NavItem>
+              </Nav>
+              <TabContent style={{ paddingTop: '10px' }}>
+                <TabPane eventKey="incoming" mountOnEnter={true} unmountOnExit={true}>
+                  <>
+                    {incomingRateGrpc.rate > 0 && (
+                      <RateTableGrpc
+                        title="GRPC Traffic (requests per second):"
+                        rate={incomingRateGrpc.rate}
+                        rateErr={incomingRateGrpc.rateErr}
+                      />
+                    )}
+                    {incomingRateHttp.rate > 0 && (
+                      <RateTableHttp
+                        title="HTTP Traffic (requests per second):"
+                        rate={incomingRateHttp.rate}
+                        rate3xx={incomingRateHttp.rate3xx}
+                        rate4xx={incomingRateHttp.rate4xx}
+                        rate5xx={incomingRateHttp.rate5xx}
+                      />
+                    )}
+                    {
+                      // We don't show a sparkline here because we need to aggregate the traffic of an
+                      // ad hoc set of [root] nodes. We don't have backend support for that aggregation.
+                    }
+                  </>
+                </TabPane>
+                <TabPane eventKey="outgoing" mountOnEnter={true} unmountOnExit={true}>
+                  <>
+                    {outgoingRateGrpc.rate > 0 && (
+                      <RateTableGrpc
+                        title="GRPC Traffic (requests per second):"
+                        rate={outgoingRateGrpc.rate}
+                        rateErr={outgoingRateGrpc.rateErr}
+                      />
+                    )}
+                    {outgoingRateHttp.rate > 0 && (
+                      <RateTableHttp
+                        title="HTTP Traffic (requests per second):"
+                        rate={outgoingRateHttp.rate}
+                        rate3xx={outgoingRateHttp.rate3xx}
+                        rate4xx={outgoingRateHttp.rate4xx}
+                        rate5xx={outgoingRateHttp.rate5xx}
+                      />
+                    )}
```

Comment:
We need to show an "Empty-State" if we don't have outgoing traffic.
Why don't we show anything for `TCP`?

Reply:
Yes, a blank tab is not very good.  I'll add the no-traffic case.

TCP data rates (as opposed to the http/grpc request rates) have been  historically been shown only via the sparklines, and not in a table format.  This PR doesn't change that in any way.  I suppose we could add TCP data rates, but if so, let's do that separately.

Reply:
fixed:

![image](https://user-images.githubusercontent.com/2104052/64450682-a904db00-d0b0-11e9-939c-07183ffc2f24.png)